spec/models/custom_button_spec.rb

```diff
@@ -384,5 +384,27 @@
         end
       end
     end
+
+    describe "publish event" do
+      context "with blank args" do
+        it "resource action calls automate_queue_hash" do
+          Timecop.freeze(Time.now.utc) do
+            expect(resource_action).to receive(:automate_queue_hash).with(vm, {}, user).and_return(:username => "foo")
+
+            User.with_user(user) { custom_button.publish_event('UI', vm) }
+          end
+        end
+      end
+
+      context "with args" do
+        it "resource action doesn't call automate_queue_hash" do
+          Timecop.freeze(Time.now.utc) do
+            expect(resource_action).not_to receive(:automate_queue_hash).with(vm, {}, user)
```

Comment:
You can drop the `with(<params>)` since you do not expect it to be called at all.